src/DIRAC/FrameworkSystem/Service/SystemAdministratorHandler.py

```diff
@@ -416,7 +417,22 @@ def _updateSoftwarePy2(self, version, rootPath, diracOSVersion):
     if rootPath and not os.path.exists(rootPath):
       return S_ERROR('Path "%s" does not exists' % rootPath)
 
-    cmdList = ['dirac-install', '-r', version, '-t', 'server']
+    if not find_executable('dirac-install'):
+      with tempfile.NamedTemporaryFile(suffix=".py", mode="wb") as installer:
```

Comment:
AFAIU you need to keep this file open, if you want to use it later. https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile
As is done above, where the `subprocess.run` is inside the `with`